src/hwProvider.ts

```diff
@@ -337,29 +360,40 @@ export class HWProvider {
         return signedTransactions;
     }
 
-    async signMessage(message: SignableMessage): Promise<SignableMessage> {
+    async signMessage(messageToSign: Message): Promise<Message> {
         if (!this.hwApp) {
             throw new ErrNotInitialized();
         }
 
-        message = this.cloneMessage(message);
-        let serializedMessage = message.serializeForSigningRaw();
-        let serializedMessageBuffer = Buffer.from(serializedMessage);
+        const message = new Message({
+            data: Buffer.from(messageToSign.data),
+            address: messageToSign.address ?? Address.fromBech32(this._account.address),
```

Comment:
Actually, I think we should always override it instead? That is, I think we should ignore `messageToSign.address`.

Reply:
Is more optimistic for page reload:
1. login
2. save address to localStorage in dApp
3. reload
4. sign message
5. => ledger device could sign but since address is no longer present in hw-provider, a re login is required